File: problems/algorithms/string-simple-substitution/ruby/code/spec/simple_substitution.test.rb
require_relative "../src/simple_substitution"

describe Text do
    describe ".substitution" do
        context "Converting message 'lorem ipsum'" do
            it "return l0r3m 1p5um" do
                expect(Text.substitution('lorem ipsum', 'aegiost', '4361057')).to eql('l0r3m 1p5um')
            end
        end

        context "Converting letters in numbers" do
            it  "return s1mpl3 m3ss4g3 3x4mpl3" do
                expect(Text.substitution('simple message example', 'aeiou', '43125')).to eql('s1mpl3 m3ss4g3 3x4mpl3')
            end
        end
    end
end

File: src/pages/exercises/string-simple-substitution/_codes/js/response/simple-substitution.test.js
import { describe, it } from 'node:test';
import assert from 'node:assert';
import { substitution } from './simple-substitution.js';

describe('String Substitution', () => {
  describe('substitution()', () => {
    it('should convert to basic leet', () => {
      assert.equal(
        substitution('lorem ipsum', 'aegiost', '4361057'),
        'l0r3m 1p5um'
      );
    });

    it('should convert to custom leet', () => {
      assert.equal(
        substitution('senha secreta', 'aegiost', '!@#$%&*'),
        '&@nh! &@cr@*!'
      );
    });
  });
});
